HW_6/task_2.js

```diff
@@ -0,0 +1,263 @@
+
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+function getEnterpriseName(id){
+    let re = null
+    enterprises.forEach(obj =>{
+        if(obj.id === id){
+            re = obj.name;            
+        }
+    });
+    return re;
+};
+
+function getEnterpriseName1(departmenName){
+    dep = null
+    enterprises.forEach(obj => {
+        obj.departments.forEach(department =>{
+            if(department.name === departmenName){
+                dep = obj.name;
+                return;                
+            }
+        });
+    });
+    
+    return dep
+}
+
+console.log(getEnterpriseName(1));
+console.log(getEnterpriseName1("Отдел маркетинга"))
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+function addEnterprise(nameEnterprise){
+    let newObj = {name: nameEnterprise, id: enterprises.length +1, departments: []};
+    enterprises.push(newObj)   
+    return enterprises
+}
+
+console.log(addEnterprise('Предприятие Тест'))
+
+// 4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+
+// Пример:
+// addDepartment(1, "Название нового отдела")
+
+function addDepartmentToEnterprise(id, nameDepartment){
+    enterprises.forEach(obj => {        
+            if(obj.id === id){
+                let newDep = {name: nameDepartment, id: obj.departments.length-1, employees_count: 0};
+                obj.departments.push(newDep);                             
+            }
+        });
+    return enterprises
+};       
+
+
+console.log(addDepartmentToEnterprise(4, 'Автотестеры'))
+
+// 5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+
+// Пример:
+// editEnterprise(1, "Новое название предприятия")
+
+function editEnterprise(id, newFirmName){
+    enterprises.forEach(obj =>{
+        if(obj.id === id){
+            obj.name = newFirmName;
+        }
+    });
+    return enterprises
+};
+
+console.log(editEnterprise(4, "Предприятие 4"))
+
+
+// 6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+
+// Пример:
+// editDepartment(7, "Новое название отдела")
+
+function editDepartment(id, newDepartment){
+    enterprises.forEach(obj =>{
+        if(obj.id === id){
+            obj.departments.name = newDepartment;             
+        };
+    });
+    return enterprises
+};
+
+console.log(editDepartment(4, "AQA"))
+
+// 7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+// Пример:
+// deleteEnterprise(1)
+
+function deleteEnterprise(id){
+    enterprises.forEach(obj =>{

```

Comment:
Don't see the purpose of `forEach` method here
You have an array with enterprises, can find enterprise in that array and delete it 
```js
            const index = enterprises.findIndex(el => el.id === enterpriseID);
            if (index !== -1) enterprises.splice(index, 1);
```